# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/CompositionTransaction.h
# Commit: d3816f82c0b9
# Full Hash: d3816f82c0b9d4ceca0bb2ad177694164ed0921f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r 876b1031a6ae -r d3816f82c0b9 editor/libeditor/CompositionTransaction.h
--- a/editor/libeditor/CompositionTransaction.h	Fri Apr 03 01:38:22 2020 +0000
+++ b/editor/libeditor/CompositionTransaction.h	Fri Apr 03 01:39:36 2020 +0000
@@ -71,14 +71,14 @@
 
   void MarkFixed();
 
-  MOZ_CAN_RUN_SCRIPT_BOUNDARY static nsresult SetIMESelection(
+  MOZ_CAN_RUN_SCRIPT static nsresult SetIMESelection(
       EditorBase& aEditorBase, dom::Text* aTextNode, uint32_t aOffsetInNode,
       uint32_t aLengthOfCompositionString, const TextRangeArray* aRanges);
 
  private:
   virtual ~CompositionTransaction() = default;
 
-  nsresult SetSelectionForRanges();
+  MOZ_CAN_RUN_SCRIPT nsresult SetSelectionForRanges();
 
   // The text element to operate upon.
   RefPtr<dom::Text> mTextNode;